### PR TITLE
Fixed a bug where scope validations where incomplete for node tests

### PR DESCRIPTION
**Pull request by chrisdp**
There was a bug where scope validations would not be run on node tests because the generated xml test components where being added to the program too late.

### CODE REVIEW

File: bsc-plugin/src/plugin.spec.ts [hunk 1]
```diff
@@ -2013,6 +2013,37 @@ describe('RooibosPlugin', () => {
         });
     });
 
+    describe('does not break class extending', () => {
+        it('does not prevent valid scope based diagnostics for node tests', () => {
```
Comment (TwitchBronBron) [inline bsc-plugin/src/plugin.spec.ts]: @chrisdp I think describe and test strings need updated to represent what this test is actually doing.

Comment (chrisdp) [reply bsc-plugin/src/plugin.spec.ts]: I thin the test string is, describe could use an update for sure.

File: bsc-plugin/src/plugin.spec.ts [hunk 2]
```diff
@@ -2013,6 +2013,37 @@ describe('RooibosPlugin', () => {
         });
     });
 
+    describe('does not break class extending', () => {
```
Comment (chrisdp) [inline bsc-plugin/src/plugin.spec.ts]:
```suggestion
    describe('does not prevent component scope validation of node tests', () => {
```

**Review by georgejecook — APPROVED** [review]

**Comment by chrisdp** [comment]
@georgejecook fixed conflicts

**Pull request merged**